### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by hortega**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,12 +20,12 @@ gen/
 /local.properties
 
 # Android Studio Files
-/.idea/workspace.xml
-/.idea/libraries
+*.iml
```
Comment (acamino) [inline .gitignore:23]: +1 for excluding IDEA specific files from CVS.

File: README.md [hunk 2]
```diff
@@ -1,18 +1,27 @@
-# twiliochat-android
+<a href="https://www.twilio.com">
+  <img src="https://static0.twilio.com/marketing/bundles/marketing/img/logos/wordmark-red.svg" alt="Twilio" width="250" />
+</a>
+
+# Twilio chat - Android
 [![Build Status](https://travis-ci.org/TwilioDevEd/twiliochat-android.svg?branch=master)](https://travis-ci.org/TwilioDevEd/twiliochat-android)
 
-Android implementation of Twilio Chat
+Learn to implement a simple chat application using Twilio Programmable Chat Client
```
Comment (acamino) [inline README.md:8]: Would you mind adding a link to the [tutorial](https://www.twilio.com/docs/tutorials/walkthrough/chat/android/java) please?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+  $ cd twiliochat-android
+  ```
+
 1. This application was developed using [Android Studio](http://developer.android.com/tools/studio/index.html).
    So if you are using a different tool like Eclipse with ADT, there might be some additional
    steps you need to follow. If you are using Android Studio just open the project
    using the IDE.
 
-1. [Twilio's IP Messaging Client](https://www.twilio.com/docs/api/ip-messaging) requires an
+1. [Twilio's Programmable Chat Client](https://www.twilio.com/docs/api/chat) requires an
```
Comment (acamino) [inline README.md:26]: +1!

File: app/build.gradle [hunk 4]
```diff
@@ -25,6 +26,8 @@ dependencies {
     compile 'joda-time:joda-time:2.9.1'
     compile 'com.mcxiaoke.volley:library-aar:1.0.1'
     testCompile 'junit:junit:4.12'
-    compile files('libs/twilio-common-android-0.3.1.jar')
-    compile files('libs/twilio-ip-messaging-android.jar')
+//    compile files('libs/twilio-common-android-0.3.1.jar')
```
Comment (acamino) [inline app/build.gradle]: Could you explain why did you left these comments?

File: app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelExtractor.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import com.twilio.twiliochat.chat.listeners.TaskCompletionListener;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ChannelExtractor {
+
+  public void extractAndSortFromChannelDescriptor(Paginator<ChannelDescriptor> paginator, final TaskCompletionListener<List<Channel>, String> listener) {
```
Comment (acamino) [inline app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelExtractor.java]: This line is too long, could we split it?

File: app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelExtractor.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      }
+
+      @Override
+      public void onError(String s) {
+        listener.onError(s);
+      }
+    });
+  }
+
+  private void extractFromChannelDescriptor(Paginator<ChannelDescriptor> paginator, final TaskCompletionListener<List<Channel>, String> listener) {
```
Comment (acamino) [inline app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelExtractor.java]: ditto - the rule applied for determine the max length of a line is how the code is shown in the docs. The maximum length is 97 chars.

File: app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelManager.java [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+          public void onSuccess(Paginator<ChannelDescriptor> channelDescriptorPaginator) {
+            extractChannelsFromPaginatorAndPopulate(channelDescriptorPaginator, listener);
+          }
+        });
+
+      }
+    });
+  }
+
+  private void extractChannelsFromPaginatorAndPopulate(final Paginator<ChannelDescriptor> channelsPaginator, final LoadChannelListener listener) {
```
Comment (acamino) [inline app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelManager.java]: ditto - long lines.

**Review by acamino — COMMENTED** [review]
Thanks @hortega! - LGTM
Just a few minor changes and we're ready.

**Comment by hortega** [comment]
Thanks @acamino , I believe I just fixed those issues

**Pull request merged**